### PR TITLE
fix + test + some cleanup

### PR DESCRIPTION
**Pull request by robert-e-davidson3**

### CODE REVIEW

File: packages/augur-sdk/src/state/logs/types.ts [hunk 1]
```diff
@@ -409,4 +409,5 @@ export interface DisputeDoc extends Log, Doc {
   tentativeWinning: boolean; // outcome is currently tentative winner
   totalRepStakedInPayout: string; // total REP across all rounds staked in completed bonds for this payout
   tentativeWinningOnRound: string; // Indicates that on a particular round this was the tentative winning payout
+  disputeRound: string; // Used in markets getter. Unsure what it is. TODO better description
```
Comment (bthaile) [inline packages/augur-sdk/src/state/logs/types.ts]: number of times the dispute bond has been filled in the reporting process. 
initial report is round 1

File: packages/augur-sdk/src/state/getter/Markets.ts [hunk 2]
```diff
@@ -1044,32 +1038,34 @@ async function getStakes(
...
+        stakeCurrent: '0',
         stakeRemaining: bondSizeCurrent,
         tentativeWinning: false,
       };
     } else {
       formattedStakeDetails[i] = {
-        outcome: outcomeValue.outcome,
-        isInvalidOutcome: outcomeValue.invalid,
-        isMalformedOutcome: outcomeValue.malformed,
+        outcome: outcomeValue.outcome || '0',
```
Comment (bthaile) [inline packages/augur-sdk/src/state/getter/Markets.ts]: shouldn't default this value it should be known.

File: packages/augur-sdk/src/state/getter/Markets.ts [hunk 3]
```diff
@@ -1044,32 +1038,34 @@ async function getStakes(
...
     if (outcomeDetails.disputeRound < market.disputeRound) {
-      const bondSizeCurrent = new BigNumber(market.totalRepStakedInMarket, 16).multipliedBy(2).minus(new BigNumber(outcomeDetails.totalRepStakedInPayout).multipliedBy(3)).toFixed();
+      const bondSizeCurrent = new BigNumber(market.totalRepStakedInMarket, 16)
+        .multipliedBy(2)
+        .minus(new BigNumber(outcomeDetails.totalRepStakedInPayout).multipliedBy(3)).toFixed();
       formattedStakeDetails[i] = {
-        outcome: outcomeValue.outcome,
-        isInvalidOutcome: outcomeValue.invalid,
-        isMalformedOutcome: outcomeValue.malformed,
+        outcome: outcomeValue.outcome || '0',
```
Comment (bthaile) [inline packages/augur-sdk/src/state/getter/Markets.ts]: same here shouldn't default this value. it should be set by `getOutcomeValue` method

**Review by bthaile — CHANGES_REQUESTED** [review]
requesting changes to set outcome when it's invalid, not defaulting to '0'

File: packages/augur-sdk/src/state/getter/Markets.ts [hunk 4]
```diff
@@ -1044,32 +1038,34 @@ async function getStakes(
...
         tentativeWinning: false,
       };
     } else {
       formattedStakeDetails[i] = {
-        outcome: outcomeValue.outcome,
-        isInvalidOutcome: outcomeValue.invalid,
-        isMalformedOutcome: outcomeValue.malformed,
+        outcome: outcomeValue.outcome || '0',
+        isInvalidOutcome: outcomeValue.invalid || false,
+        isMalformedOutcome: outcomeValue.malformed || false,
```
Comment (bthaile) [inline packages/augur-sdk/src/state/getter/Markets.ts:1070]: same with these values, if anything default in getOutcomeValue method.

File: packages/augur-sdk/src/state/getter/Markets.ts [hunk 5]
```diff
@@ -796,7 +790,7 @@ export class Markets {
       ).dividedBy(QUINTILLION);
       const settlementFee = marketCreatorFeeRate.plus(reportingFeeRate);
       // TODO: find this value from logs.
-      const noShowBondAmount = "999999000000000000000";
+      const noShowBondAmount = '999999000000000000000';
```
Comment (bthaile) [inline packages/augur-sdk/src/state/getter/Markets.ts:795]: was this added to market created log?

Comment (robert-e-davidson3) [reply packages/augur-sdk/src/state/getter/Markets.ts:795]: looks like no:
`packages/augur-core/source/contracts/Augur.sol:46`:

`    event MarketCreated(IUniverse indexed universe, uint256 endTime, string extraInfo, IMarket market, address indexed marketCreator, address designatedReporter, uint256 feePerCashInAttoCash, int256[] prices, IMarket.MarketType marketType, uint256 numTicks, bytes32[] outcomes, uint256 timestamp);
`

Comment (robert-e-davidson3) [reply packages/augur-sdk/src/state/getter/Markets.ts:795]: None of the other logs defined in Augur.sol have it either.

File: packages/augur-sdk/src/utils.ts [hunk 6]
```diff
@@ -149,17 +149,17 @@ export function calculatePayoutNumeratorsValue(
   } else {
     switch(marketType) {
       case MarketTypeName.Categorical:
-        if (!isWellFormedCategorical(payout)) return { malformed: true };
+        if (!isWellFormedCategorical(payout)) return { outcome: '0', malformed: true };
```
Comment (bthaile) [inline packages/augur-sdk/src/utils.ts]: I'm finally catching up to why outcome wasn't included. 

We need outcome set to 0 when outcome is invalid

lets not include outcome when malformed is true. We don't want to tell caller incorrect information. 

also make outcome nullable so change to `outcome?`

**Review by bthaile — CHANGES_REQUESTED** [review]
another change is needed.

File: packages/augur-sdk/src/utils.ts (hunk 6 above)
Comment (robert-e-davidson3) [reply packages/augur-sdk/src/utils.ts]: Ah. Yeah I can do that.

File: packages/augur-sdk/src/state/getter/Markets.ts (hunk 5 above)
Comment (bthaile) [reply packages/augur-sdk/src/state/getter/Markets.ts:795]: thanks

**Review by ghost — COMMENTED** [review]
This pull request has been automatically deployed to [FeaturePeek](https://featurepeek.com). 🎖

 This link will always point to this PR's latest changes: https://peek.run/2n5hl96

**Review by bthaile — APPROVED** [review]
Lgrm

**Pull request merged**